backend/src/entity/Discount.ts

```diff
@@ -0,0 +1,42 @@
+import {Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn} from "typeorm";
+import {Promotion} from "./Promotion";
+import {DiscountType} from "../data/DiscountType";
+
+/*
+* Represents a discount to a promotion. Each discount has a DiscountType which determines the discountValue.
```

Comment:
I can't remember if we talked about this 😅  but for documenting functions etc, do we want to do this for everything and go with a specific library (tsdoc, etc)?  @andrewzulaybar @ericrkuo

Reply:
yeyeye I would prefer documentation library. That way we can use annotations like `@param`, `@return`. And when you hover over it it looks nice and fancy. I can add it in this PR if its okay with @andrewzulaybar?

Reply:
@ld357 Yeah good suggestion! I'd like to have one too, probably tsdoc. @ericrkuo I just remembered you brought up https://typedoc.org/ and I forgot to comment on it -- I think we can use this if it's easy enough to work with, but I also think our project isn't big enough yet that we should be able to add these comments in manually without the need of adding another library? But maybe it might help with generating docs for the future? But what're your thoughts? Also, could you add the documentation in another PR either branching off of this one or off of master once you merge this in? It'll just be easier to review that way :)

Reply:
Oh yeah I think we can stick with manual comments like this for now! Doesn't really matter that much to me, and yup sure I can make a separate PR for it if we decide to use it